spec/services/checksum_validator_spec.rb

```diff
@@ -208,27 +208,213 @@
+            PreservedCopy::VALIDITY_UNKNOWN_STATUS
+          ].each do |initial_status|
+            it "sets status to UNEXPECTED_VERSION_ON_STORAGE_STATUS if it was previously #{initial_status}" do
+              pres_copy.status = initial_status
+              pres_copy.save!
+              expect { cv.validate_checksums }.to change { pres_copy.status }.to PreservedCopy::UNEXPECTED_VERSION_ON_STORAGE_STATUS
+              expect(pres_copy.reload.status).to eq PreservedCopy::UNEXPECTED_VERSION_ON_STORAGE_STATUS
+            end
+          end
+
+          it 'leaves status as UNEXPECTED_VERSION_ON_STORAGE_STATUS if pres copy started in that state' do
+            pres_copy.status = PreservedCopy::UNEXPECTED_VERSION_ON_STORAGE_STATUS
+            pres_copy.save!
+            expect { cv.validate_checksums }.not_to(change { pres_copy.status })
+            expect(pres_copy.reload.status).to eq PreservedCopy::UNEXPECTED_VERSION_ON_STORAGE_STATUS
+          end
+        end
+
+        context '#moab_validation_errors indicates there are structural errors' do
+          before do
+            allow(cv).to receive(:moab_validation_errors).and_return([{ Moab::StorageObjectValidator::MISSING_DIR => 'err msg' }])
+          end
+
+          [
+            PreservedCopy::OK_STATUS,
+            PreservedCopy::ONLINE_MOAB_NOT_FOUND_STATUS,
+            PreservedCopy::UNEXPECTED_VERSION_ON_STORAGE_STATUS,
+            PreservedCopy::INVALID_CHECKSUM_STATUS,
+            PreservedCopy::VALIDITY_UNKNOWN_STATUS
+          ].each do |initial_status|
+            it "sets status as INVALID_MOAB_STATUS if it was #{initial_status}" do
+              pres_copy.status = initial_status
+              pres_copy.save!
+              expect { cv.validate_checksums }.to change { pres_copy.status }.to PreservedCopy::INVALID_MOAB_STATUS
+              expect(pres_copy.reload.status).to eq PreservedCopy::INVALID_MOAB_STATUS
+            end
+          end
+
+          it 'leaves status as INVALID_MOAB_STATUS if pres copy started in that state' do
+            pres_copy.status = PreservedCopy::INVALID_MOAB_STATUS
+            pres_copy.save!
+            expect { cv.validate_checksums }.not_to(change { pres_copy.status })
+            expect(pres_copy.reload.status).to eq PreservedCopy::INVALID_MOAB_STATUS
+          end
+        end
       end
     end
 
-    context 'fails validation' do
+    context 'fails checksum validation' do
       it 'returns error codes for a pres_copy' do
         cv.validate_checksums
-        expect(cv.checksum_results.result_array.first).to have_key(:file_not_in_manifest)
+        expect(cv.results.result_array.first).to have_key(:file_not_in_manifest)
+      end
+
+      [
+        PreservedCopy::OK_STATUS,
+        PreservedCopy::ONLINE_MOAB_NOT_FOUND_STATUS,
+        PreservedCopy::INVALID_MOAB_STATUS,
+        PreservedCopy::UNEXPECTED_VERSION_ON_STORAGE_STATUS,
+        PreservedCopy::VALIDITY_UNKNOWN_STATUS
```

Comment:
I see we use this list at least twice. Do you think there's any value in factoring out a common `let` or something similar? I'm not convinced myself, but just saw the opportunity.

Reply:
yeah, totally, i had similar thoughts about that and discomfort with the repetition.

some of the lists differ in that many are 5 of the 6 possible statuses, but i'm sure some are repeated.  what might be nice is if pres copy provided a set of all possible statuses, and then we could just subtract from that as needed (e.g. `PreservedCopy.all_statuses - PreservedCopy::OK_STATUS` gets you the list you care about).

@atz was also just a bit uncomfortable with this test construction (declaring examples inside of a loop), because it can make the failed test output a bit harder to read.  not terrible, but a shared example would likely be cleaner and more idiomatic.

what if we generate another tech debt follow-on ticket or two from this thread?

Reply:
ok, filed https://github.com/sul-dlss/preservation_catalog/issues/687